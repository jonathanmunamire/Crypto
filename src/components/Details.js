const Details = () => (
  <h2>Details</h2>
);

export default Details;
