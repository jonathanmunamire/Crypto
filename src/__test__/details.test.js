import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Details from '../components/details';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
      state: {
        id: 'binance-coin',
        rank: '5',
        symbol: 'BNB',
        name: 'BNB',
        total_supply: '166801148.0000000000000000',
        ath: '732957.00000',
        price_change_24h: '',
        low_24h: '222',
        high_24h: '5555',
        market_cap_rank: '4838',
        circulating_supply: 'asdjds',
        total_volume: '38434343',
  
      },
    }),
  }));

describe('test if the page Details renders components', () => {
    test('test if the page Details is rendered', () => {
      const details = render(
            <Router>
                <Details />
            </Router>,
        );
      expect(details).toMatchSnapshot();
    });
  });