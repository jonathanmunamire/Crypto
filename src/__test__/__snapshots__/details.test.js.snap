// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if the page Details renders components test if the page Details is rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="detail_image_section"
        >
          <img
            alt="img"
            class="detail_image"
          />
        </div>
        <div
          class="detail_page"
        >
          <span>
            name:
             BNB
          </span>
          <span>
            symbol:
             BNB
          </span>
          <span>
            market cap:
             NaN
          </span>
          <span>
            price:
             NaN
          </span>
          <span>
            ATH:
             732957.00000
          </span>
          <span>
            Total Supply: 166801148.0000000000000000
          </span>
          <span>
            Circulating Supply:
             asdjds
          </span>
          <span>
            Total Volume:
             38434343
          </span>
          <span>
            Cap Rank:
             4838
          </span>
          <span>
            High 24hr:
             5555
          </span>
          <span>
            Low 24hr:
             222
          </span>
          <span>
            Price Change 24hr:
             
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="detail_image_section"
      >
        <img
          alt="img"
          class="detail_image"
        />
      </div>
      <div
        class="detail_page"
      >
        <span>
          name:
           BNB
        </span>
        <span>
          symbol:
           BNB
        </span>
        <span>
          market cap:
           NaN
        </span>
        <span>
          price:
           NaN
        </span>
        <span>
          ATH:
           732957.00000
        </span>
        <span>
          Total Supply: 166801148.0000000000000000
        </span>
        <span>
          Circulating Supply:
           asdjds
        </span>
        <span>
          Total Volume:
           38434343
        </span>
        <span>
          Cap Rank:
           4838
        </span>
        <span>
          High 24hr:
           5555
        </span>
        <span>
          Low 24hr:
           222
        </span>
        <span>
          Price Change 24hr:
           
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
